components/alert/alert-toast.js

```diff
@@ -0,0 +1,183 @@
+import '../button/button-icon.js';
+import '../button/button-subtle.js';
+import '../colors/colors.js';
+import './alert.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+
+const reduceMotion = matchMedia('(prefers-reduced-motion: reduce)').matches;
+
+const states = {
+	CLOSED: 'closed', // the toast is closed
+	CLOSING: 'closing', // the close animation is running
+	PREOPENING: 'preopening', // a pause before running the opening animation because transitions won't run when changing from 'diplay: none' to 'display: block'
+	OPENING: 'opening', // the opening animation is running
+	OPEN: 'open' // the toast is open
+};
+
+class AlertToast extends LitElement {
+
+	static get properties() {
+		return {
+			buttonText: { type: String, attribute: 'button-text' },
+			hideCloseButton: { type: Boolean, attribute: 'hide-close-button' },
+			noAutoClose: { type: Boolean, attribute: 'no-auto-close' },
+			open: { type: Boolean, reflect: true },
+			subtext: { type: String },
+			type: { type: String, reflect: true },
+			_state: { type: String, reflect: true }
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				display: block;
+			}
+
+			.d2l-alert-toast-container {
+				border-radius: 0.3rem;
+				bottom: 1.5rem;
+				box-shadow: 0 0.1rem 0.6rem 0 rgba(0,0,0,0.10);
+				display: none;
+				left: 0;
+				margin: 0 auto;
+				max-width: 600px;
+				position: fixed;
+				right: 0;
+				width: 100%;
+				z-index: 10000;
+			}
+
+			:host(:not([_state="closed"])) .d2l-alert-toast-container {
+				display: block;
+			}
+
+			:host([_state="opening"]) .d2l-alert-toast-container,
+			:host([_state="closing"]) .d2l-alert-toast-container {
+				transition-duration: 250ms;
+				transition-property: transform, opacity, visibility;
+				transition-timing-function: ease-in;
+			}
+
+			:host([_state="preopening"]) .d2l-alert-toast-container,
+			:host([_state="closing"]) .d2l-alert-toast-container {
+				opacity: 0;
+				transform: translateY(0.5rem);
+				visibility: hidden;
+			}
+
+			:host([_state="opening"]) .d2l-alert-toast-container {
+				opacity: 1;
+				transform: translateY(0);
+				visibility: visible;
+			}
+
+			.d2l-alert-toast {
+				animation: none;
```

Comment:
Looks like this is left-over from converting to transition?

Reply:
I haven't had a chance to convert `d2l-alert` to use transitions yet. This disables the `drop-in` animation that `d2l-alert` has since we don't want it running along with the toast's transitions.

Reply:
Oh I see.  Hmmm.  Icky.

Reply:
The class name sorta threw me off.  I think this would be clearer if is were:
```
d2l-alert {
    animation: none;
}
```
Then the class name could be removed from that element as well.